src/librustc_yk_sections/emit_sir.rs

```diff
@@ -349,10 +358,19 @@ impl<'a, 'tcx> ConvCx<'a, 'tcx> {
         match ty.sty {
             ty::Uint(t) => Ok(ykpack::Constant::Int(self.lower_uint(t, sclr))),
             ty::Int(t) => Ok(ykpack::Constant::Int(self.lower_int(t, sclr))),
+            ty::Bool => Ok(ykpack::Constant::Int(self.lower_bool(sclr))),
             _ => Err(()), // FIXME Not implemented.
         }
     }
 
+    fn lower_bool(&mut self, sclr: &Scalar) -> ykpack::ConstantInt {
+        match sclr {
+            Scalar::Raw{data: 0, size: 1} => ykpack::ConstantInt::from(false),
+            Scalar::Raw{data: 1, size: 1} => ykpack::ConstantInt::from(true),
+            _ => panic!("bogus boolean in MIR"),
```

Comment:
Should we improve on that error message? ;-)

Reply:
Yeah, we certainly can!